### DISCUSSION TITLE
Create a REST Controller for a Custom Entity inside plugin

### DISCUSSION THREAD

**Discussion by proudchild** [vendurehq/vendure#1183] [Q&A]
Hello All,

I'm tryin to create a custom REST Controller to expose a certain specific endpoint. The requeriment demands it to be rest.

Reading the https://www.vendure.io/docs/plugins/plugin-examples/adding-rest-endpoint/ it uses a ProductService, but what if i want to expose a custom Entity i created.

Do i have to create my VisitationService? if so, how do i do it? is there any reference im missing?

**Comment by michaelbromley** [comment]
Hi!

A "service" is just an architectural pattern, especially common in NestJS projects amongst others (Vendure is built on NestJS). All it is, is a class which contains the main business logic dealing with interacting with the database etc. They are not strictly necessary - you are free to put all that logic directly in your controller. But it is more of a convention and also can help to keep your controllers smaller and more readable.

Services in NestJS are covered here: https://docs.nestjs.com/providers#services

At the bare minimum it is a class decorated with the `@Injectable()` decorator, which allows you to inject other dependencies, most likely the TransactionalConnection so that you can fetch and modify your custom entity:

```ts
import { Injectable } from '@nestjs/common';
import { RequestContext, TransactionalConnection } from '@vendure/core';
import { Visitation } from './entities/visitation.entity';

@Injectable()
export class VisitationService {
  constructor(private connection: TransactionalConnection) {}

  findAll(ctx: RequestContext): Promise<Visitation[]> {
    return this.connection.getRepository(ctx, Visitation).find();
  }
}
```

**Reply by proudchild** [reply]
Thank you very much @michaelbromley , now that you said it it makes sense altogether, sorry for the lame question!
